### PR TITLE
Add documentation boilerplate

### PR DESCRIPTION
**Pull request by mattkram**
Adds documentation boilerplate:

* API docs generated using `sphinx-apidoc`
* Uses [`myst` parser](https://myst-parser.readthedocs.io/)
* Minimal time spent on theming, using [`pydata-sphinx-theme`](https://pydata-sphinx-theme.readthedocs.io/)

### CODE REVIEW

**Comment by ntoll** [comment]
LGTM except for the failing tests. ;-)

**Comment by mattkram** [comment]
> LGTM except for the failing tests. ;-)

Should be good now. Mind a quick run through of the instructions I just added to `CONTRIBUTING.md`?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+Or, alternately, navigate to the `docs` directory and run:
+
+```shell
+make live
+```
+
+
+Either of the above commands should launch a live dev server and you will be able to view the
+docs in your browser.
```
Comment (fpliger) [inline CONTRIBUTING.md:44]: Does it open then browser or the user needs to navigate to a specific URL?

**Review by fpliger — APPROVED** [review]
Overall LGTM. I think we need to better describe the details of adding a plugin in the user guide but can be in a follow up PR. This one is great! TY! 🚀

File: CONTRIBUTING.md (hunk 1 above)
Comment (mattkram) [reply CONTRIBUTING.md:55]: Great question! TIL, `sphinx-autobuild` as an `--open-browser` option. Now, there is a very easy way to start:

`poetry run make -C docs live`

**Comment by mattkram** [comment]
> Overall LGTM. I think we need to better describe the details of adding a plugin in the user guide but can be in a follow up PR. This one is great! TY! 🚀

Yes. This PR is definitely more about boilerplate and not content. But we should have a good skeleton to build upon now.

**Pull request merged**